web/web.go

```diff
@@ -0,0 +1,1022 @@
+func getReportFile(w http.ResponseWriter, r *http.Request) {
+	ioHelper := &conversion.IOStreams{In: os.Stdin, Out: os.Stdout}
+	var err error
+	now := time.Now()
+	filePrefix, err := getFilePrefix(now)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not get file prefix : %v", err), http.StatusInternalServerError)
+	}
+	reportFileName := "frontend/" + filePrefix + "report.txt"
+	conversion.Report(app.driver, nil, ioHelper.BytesRead, "", app.conv, reportFileName, ioHelper.Out)
+	reportAbsPath, err := filepath.Abs(reportFileName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not create absolute path : %v", err), http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(reportAbsPath))
+}
+
+type TableInterleaveStatus struct {
+	Possible bool
+	Parent   string
+	Comment  string
+}
+
+// setParentTable checks whether specified table can be
+// interleaved, if yes then it sets the Parent table for the specified
+// table and returns Parent table name, otherwise returns the issue.
+func setParentTable(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	if table == "" {
+		http.Error(w, fmt.Sprintf("Table name is empty"), http.StatusBadRequest)
+	}
+	tableInterleaveIssues := &TableInterleaveStatus{Possible: true}
+	if _, found := app.conv.SyntheticPKeys[table]; found {
+		tableInterleaveIssues.Possible = false
+		tableInterleaveIssues.Comment = "Has synthetic pk"
+	}
+	if tableInterleaveIssues.Possible == true {
+		// Search this table's foreign keys for a suitable parent table.
+		// If there are several possible parent tables, we pick the first one.
+		// TODO: Allow users to pick which parent to use if more than one.
+		for i, fk := range app.conv.SpSchema[table].Fks {
+			refTable := fk.ReferTable
+			if _, found := app.conv.SyntheticPKeys[refTable]; found {
+				continue
+			}
+			ok := checkPrimaryKeyPrefix(table, refTable, fk, tableInterleaveIssues)
+			if ok == true {
+				tableInterleaveIssues.Parent = refTable
+				sp := app.conv.SpSchema[table]
+				sp.Parent = refTable
+				sp.Fks = removeFk(sp.Fks, i)
+				app.conv.SpSchema[table] = sp
+				break
+			}
+		}
+		if tableInterleaveIssues.Parent == "" {
+			tableInterleaveIssues.Possible = false
+			tableInterleaveIssues.Comment = "No valid prefix"
+		}
+	}
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(tableInterleaveIssues)
+}
+
+func dropForeignKey(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	pos := r.FormValue("pos")
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
```

Comment:
s/spanner/Spanner/ in error message.

Reply:
changed to 
Schema is not converted or Driver is not configured properly. Please retry converting the database to Spanner.